src/coreclr/nativeaot/Test.CoreLib/src/System/Runtime/CompilerServices/ClassConstructorRunner.cs

```diff
@@ -66,13 +66,6 @@ private static unsafe void CheckStaticClassConstruction(ref StaticClassConstruct
 
                     ((delegate*<void>)oldInitializationState)();
 
-                    // Insert a memory barrier here to order any writes executed as part of static class
-                    // construction above with respect to the initialized flag update we're about to make
-                    // below. This is important since the fast path for checking the cctor uses a normal read
-                    // and doesn't come here so without the barrier it could observe initialized == 1 but
-                    // still see uninitialized static fields on the class.
-                    Interlocked.MemoryBarrier();
```

Comment:
There is a copy of this in CoreLib version of this. I think should also be removed.